### PR TITLE
[17.0][MIG] contract_payment_mode: Migration to 17.0

### PR DESCRIPTION
**Pull request by Tisho99**
Standard migration

Added little improvements requested by  Rousseldenis

Supersedes #1088

T-6501

### CODE REVIEW

**Comment by Tisho99** [comment]
@HaraldPanten @ValentinVinagre

**Comment by HaraldPanten** [comment]
@pedrobaeza could you assign this migration to us, and close the other migration PR?

@rousseldenis Could you help us with the reviews? THX!

**Review by rousseldenis — APPROVED** [review]
Code review

File: contract_payment_mode/readme/USAGE.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+1.  Go to *Invoicing \> Customers \> Customer Contracts*.
+2.  Create one.
+3.  Select a partner to which invoice.
+4.  If the partner has a payment mode, this payment mode is selected
+    here.
+5.  If not, or if you want another payment mode, you can change it in
+    the corresponding field.
+6.  Click on **Generate recurring invoices automatically** checkbox.
```
Comment (HaraldPanten) [inline contract_payment_mode/readme/USAGE.md]: Is step number 6 correct?

**Review by HaraldPanten — CHANGES_REQUESTED** [review]

File: contract_payment_mode/models/contract.py [hunk 2]
```diff
@@ -9,15 +9,19 @@ class ContractContract(models.Model):
...
     )
 
-    @api.onchange("partner_id")
-    def on_change_partner_id(self):
-        partner = self.with_company(self.company_id).partner_id
-        if self.contract_type == "purchase":
-            self.payment_mode_id = partner.supplier_payment_mode_id.id
-        else:
-            self.payment_mode_id = partner.customer_payment_mode_id.id
+    @api.depends("partner_id")
```
Comment (ValentinVinagre) [inline contract_payment_mode/models/contract.py]: Shouldn't it also depend on the 'contract_type' field?

Comment (Tisho99) [reply contract_payment_mode/models/contract.py]: Yes, it should, changed

File: contract_payment_mode/readme/USAGE.md (hunk 1 above)
Comment (Tisho99) [reply contract_payment_mode/readme/USAGE.md]: No, it is not needed. Step removed

**Comment by Tisho99** [comment]
@ValentinVinagre @HaraldPanten 

I updated the PR to your requested changes

**Review by HaraldPanten — APPROVED** [review]
@pedrobaeza could we merge this one?

THX!

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**